apm-agent-plugins/apm-apache-httpclient-plugin/src/main/java/co/elastic/apm/agent/httpclient/LegacyApacheHttpClientInstrumentation.java

```diff
@@ -56,7 +56,7 @@
     public static class LegacyApacheHttpClientAdvice {
         @Nullable
         @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
-        public static Object onBeforeExecute(@Advice.Argument(0) HttpHost host,
+        public static Object onBeforeExecute(@Advice.Argument(0) @Nullable HttpHost host,
                                              @Advice.Argument(1) HttpRequest request) {
```

Comment:
why not annotate `HttpRequest` with `@Nullable` for consistency, while there is an `instanceof` check, it would make reading it slightly easier.

Reply:
Consistency shouldn't be a consideration here, only whether it can be `null`. Do you have a reason to think it may be?

Reply:
Not really, let's keep it this way then.